docs/root/intro/arch_overview/advanced/matching/matching_listener.rst

```diff
@@ -0,0 +1,63 @@
+.. _arch_overview_matching_listener:
+
+Matching Filter Chains in Listeners
+===================================
+
+Envoy listeners implement the :ref:`matching API <envoy_v3_api_msg_.xds.type.matcher.v3.Matcher>` for selecting a filter
+chain based on a collection of :ref:`network inputs <extension_category_envoy.matching.network.input>`. Matching is done
+once per connection. Connections are drained when the associated named filter chain configuration changes, but not when
+the filter chain matcher is the only updated field in a listener.
+
+The action in the matcher API must be a string value corresponding to the name of the filter chain. If there is no
+filter chain with the given name, the match fails, and the :ref:`default filter chain
+<envoy_v3_api_field_config.listener.v3.Listener.default_filter_chain>` is used if specified, or the connection is
+rejected. Filter chain matcher requires that all filter chains in a listener are uniquely named.
+
+The matcher API replaces the existing filter :ref:`filter_chain_match
+<envoy_v3_api_field_config.listener.v3.FilterChain.filter_chain_match>` field. When using the matcher API, the filter
+chain match field is ignored and should not be set.
```

Comment:
Should this be exposed to the user in some way? ie fail the config or at the very least log something?

Reply:
Added a debug log. Failing might complicate migration IMHO, so just ignoring the field seems reasonable as it is an opt-in feature.